File: erlang/4.4/powers.erl
-module(powers).
-export([raise/2]).

raise(_X, 0) ->
		1;
raise(X, 1) ->
		X;
raise(X, N) when N > 0 ->
		raise(X, N, 1);
raise(X, N) when N < 0 ->
		1.0 / (raise(X, -1 * N)).


raise(_X, 0, Acc) ->
		Acc;
raise(X, N, Acc) ->
		raise(X, N - 1, X * Acc).

File: elixir/4.4/powers.ex
defmodule Powers do
  import Kernel, except: [raise: 2]
  
  def raise(_, 0) do
    1
  end

  def raise(x, 1) do
    x
  end

  def raise(x, n) when n > 0 do
    raise(x, n, 1)
  end

  def raise(x, n) when n < 0 do
    1.0 / raise(x, -1 * n)
  end

  defp raise(x, 0, acc) do 
    acc
  end
  
  defp raise(x, n, acc) do
    raise(x, n - 1, acc * x)
  end
end
